app/controllers/media_controller.rb

```diff
@@ -0,0 +1,116 @@
+class MediaController < ApplicationController
+  def index
+    @movies = Movie.order('votes DESC').first(10)
+    @books = Book.order('votes DESC').first(10)
+    @albums = Album.order('votes DESC').first(10)
+  end
+
+  def all
+    medium_type = params[:type]
+    model_class = type_to_model(medium_type)
+    @media = model_class.order('votes DESC')
+    @medium_type_name = medium_type.capitalize
+
+  end
+
+  def show
+    @medium_id = params[:id].to_i
+    @medium = Medium.find_by_id(@medium_id)
```

Comment:
We use this bit of code at the top of multiple controller actions (`show`, `edit`, `update`, `upvote`, and we could use it in `destroy`), so we should try to refactor this into a controller filter with `before_action`.